### PR TITLE
add cpu adamW

### PR DESCRIPTION
**Pull request by Gy-Lu**
Added cpu AdamW.
Separate the class `CPUAdam` and `CPUAdamW`, users can not defined an adamw by passing a parameter `adamw_mode` when init.

### CODE REVIEW

File: colossalai/nn/optimizer/cpu_adamw.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
+import torch
+import math
+
+
+class CPUAdamW(torch.optim.Optimizer):
+    optimizer_id = 0
+
```
Comment (ver217) [inline colossalai/nn/optimizer/cpu_adamw.py:7]: Can we set `num_fp32_shards_per_param` to support hybrid optimization?

File: colossalai/nn/optimizer/cpu_adamw.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+                if target_device.type == 'cpu':
+                    assert state['exp_avg'].device.type == 'cpu', "exp_avg should stay on cpu"
+                    assert state['exp_avg_sq'].device.type == 'cpu', "exp_avg should stay on cpu"
+                    self.cpu_adam_op.adam_update(self.opt_id, state['step'], group['lr'], beta1, beta2, group['eps'],
+                                                 group['weight_decay'], group['bias_correction'], p.data, p.grad.data,
+                                                 state['exp_avg'], state['exp_avg_sq'], self.loss_scale)
+                elif target_device.type == 'cuda':
+                    # FIXME() prepare grad on cuda
+                    if p.grad.device.type == 'cpu':
```
Comment (ver217) [inline colossalai/nn/optimizer/cpu_adamw.py:109]: You can remove this.

File: colossalai/nn/optimizer/cpu_adamw.py (hunk 1 above)
Comment (Gy-Lu) [reply colossalai/nn/optimizer/cpu_adamw.py:7]: OK, I forget to update the code.

File: colossalai/nn/optimizer/cpu_adamw.py [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        except ImportError:
+            raise ImportError('Please install colossalai from source code to use CPUAdam')
+        self.cpu_adam_op = cpu_adam
+        self.cpu_adam_op.create_adam(self.opt_id, lr, betas[0], betas[1], eps, weight_decay, adamw_mode, simd_log)
+
+    def __del__(self):
+        if self.cpu_adam_op:
+            self.cpu_adam_op.destroy_adam(self.opt_id)
+
+    def torch_adamw_update(self,
```
Comment (ver217) [inline colossalai/nn/optimizer/cpu_adamw.py:40]: Is it possible for us to use fused `Adam` here? @feifeibear @Gy-Lu

Comment (Gy-Lu) [reply colossalai/nn/optimizer/cpu_adamw.py:40]: ok, I'll try :)

File: colossalai/nn/optimizer/cpu_adamw.py (hunk 2 above)
Comment (Gy-Lu) [reply colossalai/nn/optimizer/cpu_adamw.py:109]: Done

File: colossalai/nn/optimizer/cpu_adamw.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
+import torch
+import math
+
+
+class CPUAdamW(torch.optim.Optimizer):
```
Comment (feifeibear) [inline colossalai/nn/optimizer/cpu_adamw.py:5]: why not derive from CPUAdam?
Or CPUAdam and CPUAdamW both derive from CPUAdamBase...
A lot of duplicated code between CPUAdam and CPUAdamW

Comment (Gy-Lu) [reply colossalai/nn/optimizer/cpu_adamw.py:5]: Thanks for the advice, I'll consider it.

**Pull request closed without merging**